src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -153,69 +175,68 @@ class NotificationService(
-                    CommunicationPreference.EMAIL -> {
-                        notifyClient.sendEmail(NotificationType.EMAIL_SUMMARY.value, userPreference.email, generateTemplateValues(chunk, communicationPreference), null)
+        if (!userHasSnoozedNotifications(userPreference.snoozeUntil)) {
+            // Only send the latest notification for a shift if there are multiple
+            notificationGroup
+                    .asSequence()
+                    .groupBy { it.compoundKey() }
+                    .map { (_, value) -> value.maxBy { it.shiftModified } }.filterNotNull()
+                    .sortedBy { it.shiftDate }.chunked(10)
+                    .forEach { chunk ->
+                        log.info("Sending ${chunk.size} deduplicated notifications to ${userPreference.quantumId}, preference set to ${userPreference.commPref}")
+
+                        when (val communicationPreference = CommunicationPreference.from(userPreference.commPref)) {
+                            CommunicationPreference.EMAIL -> {
+                                notifyClient.sendEmail(NotificationType.EMAIL_SUMMARY.value, userPreference.email, generateTemplateValues(chunk, communicationPreference), null)
+                                log.info("Sent Email notification (${chunk}.size} lines) for $quantumId")
+                            }
+                            CommunicationPreference.SMS -> {
+                                notifyClient.sendSms(NotificationType.SMS_SUMMARY.value, userPreference.sms, generateTemplateValues(chunk, communicationPreference), null)
+                                log.info("Sent SMS notification (${chunk}.size} lines) for $quantumId")
+                            }
+                            else -> {
+                                log.info("Skipped sending notifications for ${userPreference.quantumId}, preference ${userPreference.commPref}")
+                            }
+                        }
                     }
-                    CommunicationPreference.SMS -> {
-                        notifyClient.sendSms(NotificationType.SMS_SUMMARY.value, userPreference.sms, generateTemplateValues(chunk, communicationPreference), null)
-                    }
-                    else -> {
-                        log.info("Skipping sending notifications for ${userPreference.quantumId}")
-                    }
-                }
-            }
+        } else {
+            log.info("Skipped sending notifications to ${userPreference.quantumId}, snooze set to ${userPreference.snoozeUntil}")
         }
     }
 
-    private fun shouldSend(userPreference: UserPreference): Boolean {
-        val isNotSnoozed = (userPreference.snoozeUntil == null || userPreference.snoozeUntil != null && userPreference.snoozeUntil!!.isBefore(LocalDate.now(clock)))
-        val isValidCommunicationMethod = when (userPreference.commPref) {
```

Comment:
This was for before, when the user had two columns, email and sms, and useEmail, and useSms. we validate when this data is set now, so we don't need to check here anymore.